ground-server/src/components/actuation-box.tsx

```diff
@@ -15,50 +16,84 @@ import {
 import { Button } from "@/components/ui/button";
 import { sendCommand } from "@/lib/grpcClient";
 import { type Command } from "@/proto-out/command";
+import { useToast } from "@/components/ui/use-toast";
+
+interface ButtonConfig {
+  label: string;
+  stateLabel: string;
+  command: Command;
+  isOn: boolean; // Indicates if this button represents an "on" or "off" state
+}
 
 interface ActuationBoxProps {
   title: string;
-  onButtonLabel: string;
-  onStateLabel: string;
-  onCommand: Command;
-  offButtonLabel: string;
-  offStateLabel: string;
-  offCommand: Command;
+  buttons: ButtonConfig[];
   initialStateLabel: string;
+  useSwitch?: boolean; // Introduced the useSwitch prop
+  switchLabel?: string;
+  switchOnCommand?: Command;
+  switchOffCommand?: Command;
 }
 
 export default function ActuationBox({
   title,
-  onButtonLabel,
-  onStateLabel,
-  onCommand,
-  offButtonLabel,
-  offStateLabel,
-  offCommand,
+  buttons,
   initialStateLabel,
+  useSwitch = true,
+  switchLabel = "",
+  switchOnCommand,
+  switchOffCommand,
 }: ActuationBoxProps) {
   const [currentState, setCurrentState] = useState<string | null>(null);
+  const [isOnState, setIsOnState] = useState<boolean | null>(null); // Track whether the current state is "on" or "off"
+  const { toast } = useToast();
+  const [isSwitchOn, setIsSwitchOn] = useState<boolean>(true); // State for the switch
+
+  const handleButtonClick = (button: ButtonConfig) => {
+    if (!isSwitchOn && useSwitch === false) {
+      // If switch is used and it's off, disable button actions
```

Comment:
I don't think we want this. We want to be able to switch the Ball Valve's direction before we turn it back on